File: ios/Classes/map/model/UnifiedRoutePOISearchResult.m
//
// Created by Yohom Bao on 2018-12-04.
//

#import <AMapSearch/AMapSearchKit/AMapSearchObj.h>
#import <AMapSearch/AMapSearchKit/AMapCommonObj.h>
#import "UnifiedRoutePOISearchResult.h"
#import "UnifiedAMapOptions.h"
#import "NSArray+Rx.h"
#import "AMapGeoPoint+LatLng.h"


@implementation UnifiedRoutePOISearchResult {

}
- (instancetype)initWithAMapRoutePOISearchResponse:(AMapRoutePOISearchResponse *)result {
    if ([self init]) {
        _query = nil;
        _routePoiList = [result.pois map:^(AMapPOI *poi) {
            return [[UnifiedRoutePOIItem alloc] initWithAMapPOI:poi];
        }];
    }
    return self;
}

@end

@implementation UnifiedRoutePOIItem {

}
- (instancetype)initWithAMapPOI:(AMapPOI *)aMapPOI {
    if ([self init]) {
        _id = aMapPOI.uid;
        _title = aMapPOI.name;
        _point = [aMapPOI.location toLatLng];
        _distance = aMapPOI.distance;
        _duration = 0;
    }
    return self;
}

@end

File: android/src/main/kotlin/me/yohom/amapbase/search/model/UnifiedRoutePoiSearchResult.kt
package me.yohom.amapbase.search.model

import com.amap.api.services.routepoisearch.RoutePOIItem
import com.amap.api.services.routepoisearch.RoutePOISearchQuery
import com.amap.api.services.routepoisearch.RoutePOISearchResult
import me.yohom.amapbase.search.toLatLng

class UnifiedRoutePoiSearchResult(routePOISearchResult: RoutePOISearchResult) {
    private val routePoiList: List<UnifiedRoutePOIItem> = routePOISearchResult.routePois?.map { UnifiedRoutePOIItem(it) }
            ?: listOf()
    private val query: RoutePOISearchQuery = routePOISearchResult.query
}

private class UnifiedRoutePOIItem(poiItem: RoutePOIItem) {
    val id: String = poiItem.id
    val title: String = poiItem.title
    val point: LatLng? = poiItem.point?.toLatLng()
    val distance: Float = poiItem.distance
    val duration: Float = poiItem.duration
}